zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirector.java

```diff
@@ -225,6 +229,49 @@ private void removeExporter(final String exporterId, final ExporterContainer con
     }
   }
 
+  /**
+   * Enables an exporter with the given id and descriptor. The exporter will start exporting records
+   * after this operation completes.
+   *
+   * <p>It is expected that the exporter to initialize from the metadata is of same type as the
+   * exporter to enable. The caller of this method must verify that.
+   *
+   * @param exporterId id of the exporter to enable
+   * @param initializationInfo the info required to initialize the exporter state
+   * @param descriptor the descriptor of the exporter to enable
+   * @return future which will be completed after the exporter is enabled.
+   */
+  public ActorFuture<Void> enableExporter(
+      final String exporterId,
+      final ExporterInitializationInfo initializationInfo,
+      final ExporterDescriptor descriptor) {
+    if (actor.isClosed()) {
+      return CompletableActorFuture.completed(null);
+    }
+
+    return actor.call(() -> addExporter(exporterId, initializationInfo, descriptor));
+  }
+
+  private void addExporter(
+      final String exporterId,
+      final ExporterInitializationInfo initializationInfo,
+      final ExporterDescriptor descriptor) {
+    final ExporterContainer container =
+        new ExporterContainer(descriptor, partitionId, initializationInfo);
+    container.initContainer(actor, metrics, state, exporterPhase);
+    try {
+      container.configureExporter();
+    } catch (final Exception e) {
+      LOG.error("Failed to configure exporter '{}'", exporterId, e);
+      LangUtil.rethrowUnchecked(e);
+    }
+    // initializes metadata and position in the runtime state
+    container.initMetadata();
+    container.openExporter();
+    containers.add(container);
+    LOG.info("Exporter '{}' is enabled.", exporterId);
```

Comment:
:question: Will this be printed on every start up, or only when the exporter goes from disabled to enabled? Because I guess that's the only time we care about this? :thinking:

Reply:
Only when it goes from disabled to enabled.

Reply:
Perfect, thanks! :+1: